services/src/main/java/org/keycloak/protocol/oidc/mappers/LightWeightAccessTokenMapper.java

```diff
@@ -0,0 +1,75 @@
+package org.keycloak.protocol.oidc.mappers;
+import org.keycloak.common.Profile;
+import org.keycloak.common.util.Time;
+import org.keycloak.models.ClientSessionContext;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.ProtocolMapperModel;
+import org.keycloak.models.SingleUseObjectProvider;
+
+import org.keycloak.models.UserSessionModel;
+import org.keycloak.provider.EnvironmentDependentProviderFactory;
+import org.keycloak.provider.ProviderConfigProperty;
+import org.keycloak.representations.AccessToken;
+
+import java.util.ArrayList;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.keycloak.representations.IDToken.SESSION_ID;
+import static org.keycloak.representations.JsonWebToken.AZP;
+
+public class LightWeightAccessTokenMapper extends AbstractOIDCProtocolMapper implements OIDCAccessTokenMapper, EnvironmentDependentProviderFactory {
+    public static String LIGHT_WEIGHT_ACCESS_TOKEN_PROVIDER_ID = "oidc-light-weight-access-token-mapper";
+    private static final List<ProviderConfigProperty> configProperties = new ArrayList<>();
+    @Override
+    public String getDisplayCategory() {
+        return TOKEN_MAPPER_CATEGORY;
+    }
+
+    @Override
+    public String getDisplayType() {
+        return "Light weight access token";
+    }
+
+    @Override
+    public String getHelpText() {
+        return "Delete the information contained by default in the access token. " +
+                "The deleted information can be obtained at the token introspection endpoint.";
+    }
+
+    @Override
+    public List<ProviderConfigProperty> getConfigProperties() {
+        return configProperties;
+    }
+
+    @Override
+    public String getId() {
+        return LIGHT_WEIGHT_ACCESS_TOKEN_PROVIDER_ID;
+    }
+
+    @Override
+    public AccessToken transformAccessToken(AccessToken token, ProtocolMapperModel mappingModel, KeycloakSession session,
```

Comment:
I agree.
I am reffering to [JSON Web Token (JWT) Profile for OAuth 2.0 Access Tokens documentation](https://datatracker.ietf.org/doc/html/rfc9068#name-data-structure). However, this discussion is out of scope of this PR.

Reply:
@cgeorgilakis @mposolda I think that the intention of introducing `LightWeightAccessTokenMapper` is to hide such claims from an access token that a protocol mapper cannot control.

Also it seems that you are worry about putting a lot of information onto `SingleUseObjectProvider`.

IMO, the following claims are only available when Keycloak receives a token request from a client and generate an access token:
-   scope, nonce, jti, iat, exp, client_id, azp, session_state, iss(realm name), cnf (OAuth MTLS, DPoP) 

Therefore, if we want not to put such claims onto an access token but want to put them onto a token introspection response, we need to keep them by some mechanism like `SingleUseObjectProvider`.

One option of a lightweight access token is that we must put all such claims to an access token. Another option is that we only put some of such claims to an access token and others are kept by using `SingleUseObjectProvider` by this `LightWeightAccessTokenMapper` . WDYT?

Reply:
I see the idea behind the mapper but I don't think we need it at all. 

For me, most of the claims being removed by this method are a key to accessing protected resources on a resource server.  We can't be opinionated on which claims should go into the access token as it is use-case-specific. For some, you might need the subject in a lightweight token when accessing protected resources, or even the client to which the token was issued.

We should keep in mind that a lightweight token is just about making sure you have minimal claims to avoid introspecting tokens. The more claims you remove, the more your access token becomes a *reference token* instead, for which you are forced to introspect the token in order to resolve additional information at the resource server when processing bearer tokens.

Reply:
> I am reffering to [JSON Web Token (JWT) Profile for OAuth 2.0 Access Tokens documentation](https://datatracker.ietf.org/doc/html/rfc9068#name-data-structure).

You have a good point here and we should probably choose the basic claims a lightweight token should have based on this spec. It already provides the minimal set (similar to ID Token) of claims and that spec was based on feedback from different vendors and use cases.